source/usb/usb_lib.c

```diff
@@ -2281,11 +2438,87 @@ const struct {
 #if (USBD_HID_ENABLE)
     USBD_STR_VAL(HID_STRDESC),
 #endif
+#if (USBD_WEBUSB_ENABLE)
+    USBD_STR_VAL(WEBUSB_STRDESC),
+#endif
 #if (USBD_MSC_ENABLE)
     USBD_STR_VAL(MSC_STRDESC),
 #endif
 };
 
+#if (USBD_WEBUSB_ENABLE)
+
+#define WEBUSB_NUM_FUNCTIONS              (1)
+
+#define WEBUSB_WTOTALLENGTH               (WEBUSB_DESCRIPTOR_SET_HEADER_SIZE +      \
+                                           WEBUSB_CONFIGURATION_SUBSET_HEADER_SIZE +\
+                                           (WEBUSB_NUM_FUNCTIONS * (WEBUSB_FUNCTION_SUBSET_HEADER_SIZE + 1)))
+
+__weak \
+const U8 USBD_WebUSBAllowedOriginsHeader[] = {
```

Comment:
You may want to remove the allowed origins descriptor, since it was dropped from the WebUSB spec. It's benign to leave it in, but it's less code to maintain.

Reply:
You are correct. It's no longer in specification `0x01` is now marked as reserved and `0x02` is used to get url. I'm removing it.